app/src/main/java/com/yacgroup/yacguide/AscendActivity.kt

```diff
@@ -58,17 +59,22 @@ class AscendActivity : AppCompatActivity() {
 
         // The database query is different from null,
         // if this activity is called from the TourbookActivity or TourbookAscendActivity.
-        _ascend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY, DatabaseWrapper.INVALID_ID))
-                ?: Ascend().apply {
-                    partnerIds = ArrayList()
-                    notes = ""
-                }
+        _outdatedAscend = _db.getAscend(intent.getIntExtra(IntentConstants.ASCEND_KEY, DatabaseWrapper.INVALID_ID))
         // The intent constant ROUTE_KEY is only available, if this activity is called from
         // the route DescriptionActivity.
-        val routeId = intent.getIntExtra(IntentConstants.ROUTE_KEY, DatabaseWrapper.INVALID_ID)
-        _route = _db.getRoute(routeId.takeUnless { it == DatabaseWrapper.INVALID_ID }
-                ?: _ascend.routeId)
+        _route = _db.getRoute(_outdatedAscend?.routeId
+                    ?: intent.getIntExtra(IntentConstants.ROUTE_KEY, DatabaseWrapper.INVALID_ID))
         // Beware: _route may still be null (if the route of this ascend has been deleted meanwhile)
+        _ascend = Ascend().apply {
+                    id = _outdatedAscend?.id ?: 0
+                    routeId = _outdatedAscend?.routeId ?: _route!!.id
+                    styleId = _outdatedAscend?.styleId ?: 0
+                    year = _outdatedAscend?.year ?: 0
+                    month = _outdatedAscend?.month ?: 0
+                    day = _outdatedAscend?.day ?: 0
+                    partnerIds = _outdatedAscend?.partnerIds ?: ArrayList()
+                    notes = _outdatedAscend?.notes ?: ""
+                }
```

Comment:
You initialize all properties again, however the `Ascend` class does it already for some of them. I think a more consistent way would be to do it for just the required ones.

What returns `_route!!.id`, if `_route` is `null`?

Reply:
1. `_route` can only be `null` if `_outdatedAscend` is not `null`. So no problem here: However, I guess it's again time to have some unit tests :)
2. Yes, this initialization of `_ascend` is needed because I want to copy `_outdatedAscend` and there is no `copy` method in Kotlin for non-`data`-classes